### PR TITLE
BS-14360 Handle active organization / repository

### PR DESCRIPTION
**Pull request by rbioteau**
* Instead of storing a global pref for active organization use a projec
scope IEclipsePreference

https://github.com/bonitasoft/bonita-studio-sp/pull/108

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/OrganizationLabelProvider.java [hunk 1]
```diff
@@ -56,9 +58,9 @@ public String getText(Object element) {
 
 	@Override
 	public void update(ViewerCell cell) {
-		String orgaName = getText(cell.getElement());
-		String activeOrganization = BonitaStudioPreferencesPlugin.getDefault().getPreferenceStore().getString(ActorsPreferenceConstants.DEFAULT_ORGANIZATION) ;
-		StyledString styledString = new StyledString();
+		final String orgaName = getText(cell.getElement());
+        final String activeOrganization = getActiveOrganization();
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/OrganizationLabelProvider.java]: issue with indentation

File: bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/OrganizationLabelProvider.java [hunk 2]
```diff
@@ -68,6 +70,12 @@ public void update(ViewerCell cell) {
 		cell.setStyleRanges(styledString.getStyleRanges());
 	}
 
+    private String getActiveOrganization() {
+        final IScopeContext projectScope = RepositoryManager.getInstance().getCurrentRepository().getScopeContext();
+        return projectScope.getNode(ActorsPlugin.PLUGIN_ID).get(ActorsPreferenceConstants.DEFAULT_ORGANIZATION,
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/OrganizationLabelProvider.java]: Sure that the first getNode never return null? If no, check for null and return the ActorsPreferenceConstants.DEFAULT_ORGANIZATION_NAME

File: bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/SelectOrganizationWizardPage.java [hunk 3]
```diff
@@ -105,6 +106,12 @@ public String getText(final Object element) {
         setControl(mainComposite);
     }
 
+    private String getActiveOrganization() {
+        final IScopeContext projectScope = RepositoryManager.getInstance().getCurrentRepository().getScopeContext();
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/SelectOrganizationWizardPage.java]: duplicated code with OrganizationLabelProvider.getActiveOrganization()

File: tests/org.bonitasoft.studio.test.swtbot.suite/SVN Test Suite.launch [hunk 4]
```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (apupier) [inline tests/org.bonitasoft.studio.test.swtbot.suite/SVN Test Suite.launch]: sure that you want to commit this launch configuration in this PR?

**Comment by apupier** [comment]
"fix version" is not specified on bugtracker, maybe check with Duy if we do it on 7.1.x or master.

File: bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/OrganizationLabelProvider.java (hunk 2 above)
Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.actors/src/org/bonitasoft/studio/actors/ui/wizard/page/OrganizationLabelProvider.java]: sure. I check source and node is created if it doesn't exist

**Comment by rbioteau** [comment]
Duy is ok for 7.1.4

File: bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/Repository.java [hunk 5]
```diff
@@ -733,4 +735,9 @@ public boolean isLoaded() {
         return stores != null && !stores.isEmpty();
     }
 
+    public IScopeContext getScopeContext() {
+        return new ProjectScope(project);
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/Repository.java]: extra empty line

**Comment by apupier** [comment]
the default User used to update the organization shoudl be stored also per project scope otherwise it fails when switching repository if this User oesn't exist in the another organization

File: bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/Repository.java [hunk 6]
```diff
@@ -161,16 +163,11 @@ public Repository create(final IProgressMonitor monitor) {
...
-                addNature("org.bonitasoft.studio.common.repository.bonitaNature").
-                addNature(JavaCore.NATURE_ID).
-                addNature("org.eclipse.pde.PluginNature").
-                addNature("org.eclipse.jdt.groovy.core.groovyNature").
-                addBuilder("org.eclipse.jdt.core.javabuilder").
-                addBuilder("org.eclipse.xtext.ui.shared.xtextBuilder").
-                addBuilder("org.eclipse.pde.ManifestBuilder").
-                addBuilder("org.eclipse.pde.SchemaBuilder");
+        return new CreateBonitaBPMProjectOperation(workspace, projectName).addNature("org.eclipse.xtext.ui.shared.xtextNature")
+                .addNature("org.bonitasoft.studio.common.repository.bonitaNature").addNature(JavaCore.NATURE_ID).addNature("org.eclipse.pde.PluginNature")
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/Repository.java]: previous formatting was more readable (and don't we have some constants for al these values?)

File: bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/Repository.java [hunk 7]
```diff
@@ -701,16 +698,11 @@ public void migrate(final IProgressMonitor monitor) throws CoreException, Migrat
...
-                addNature(JavaCore.NATURE_ID).
-                addNature("org.eclipse.pde.PluginNature").
-                addNature("org.eclipse.jdt.groovy.core.groovyNature").
-                addBuilder("org.eclipse.jdt.core.javabuilder").
-                addBuilder("org.eclipse.xtext.ui.shared.xtextBuilder").
-                addBuilder("org.eclipse.pde.ManifestBuilder").
-                addBuilder("org.eclipse.pde.SchemaBuilder");
+        return new BonitaBPMProjectMigrationOperation(project, this).addNature("org.eclipse.xtext.ui.shared.xtextNature")
+                .addNature("org.bonitasoft.studio.common.repository.bonitaNature").addNature(JavaCore.NATURE_ID).addNature("org.eclipse.pde.PluginNature")
+                .addNature("org.eclipse.jdt.groovy.core.groovyNature").addBuilder("org.eclipse.jdt.core.javabuilder")
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/Repository.java]: previous formatting war more readable (and don't we have some constants for al these values?)

Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/Repository.java]: Just use the automatic formatting from BOS-formatter.xml

**Pull request merged**